spec/classes/lightblue/application/datamgmt_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'spec_helper'
+
+describe 'lightblue::application::datamgmt' do
+  let(:hiera_config){ 'spec/fixtures/hiera/hiera.yaml' }
+
+  let :facts do
+    {
+      :architecture => 'x86_64',
+      :osfamily => 'RedHat',
+      :operatingsystemrelease => '7.0'
+    }
+  end
+
+  context 'with default params' do
+    it do
+      should contain_package('lightblue-data-mgmt') \
+        .with_ensure('latest') \
+        .with_require(['Class[Lightblue::Yumrepo::Lightblue]', 'Class[Lightblue::Eap]'])
+    end
+
+    it do
```

Comment:
Sounds like if this is best practice it could be incorporated into the testing framework by collecting failures before reporting an issue.  I don't want to come off as saying "but that's not how it's done in any other unit tests" but it isn't.  Asserts in junit fail immediately and you don't see if anything further down the line would have failed.  Is this because it's harder in other languages to re-run a test and do separate asserts?  Maybe.  Looks like this framework makes it pretty simple to do.

We could write guidance such as "if an assertion cannot pass if another assertion fails those related assertions must be inside the same test, else it's a matter of preference" but I don't know how we want to go.  I'm a fan of being consistent.

@dcrissman you've done more of these tests than any of us, what do you think?